--- tiff-3.7.1.orig/libtiff/tif_getimage.c	2004-12-29 12:54:15.146006632 +0100
+++ tiff-3.7.1/libtiff/tif_getimage.c	2004-12-29 12:54:55.329897752 +0100
@@ -247,7 +247,7 @@
     TIFFGetFieldDefaulted(tif, TIFFTAG_SAMPLESPERPIXEL, &img->samplesperpixel);
     TIFFGetFieldDefaulted(tif, TIFFTAG_EXTRASAMPLES,
 	&extrasamples, &sampleinfo);
-    if (extrasamples > 1)
+    if (extrasamples >= 1)
     {
 	switch (sampleinfo[0]) {
 	case EXTRASAMPLE_UNSPECIFIED:	/* Workaround for some images without */
