--- tiff-v3.5.6-beta/libtiff/Makefile.in.orig	Thu Jun 14 20:29:40 2001
+++ tiff-v3.5.6-beta/libtiff/Makefile.in	Thu Jun 14 20:54:34 2001
@@ -65,7 +65,7 @@
 OPTIMIZER=-O
 CFLAGS	= @ENVOPTS@ @LIBCOPTS@ ${COPTS} ${OPTIMIZER} ${IPATH} ${CONF_LIBRARY}
 SONAME  = libtiff.so.3
-SHLIB   = libtiff.so.3.5
+SHLIB   = libtiff.@DSOSUF_VERSION@
 #
 SRCS	= \
 	tif_aux.c \
@@ -151,8 +151,7 @@
 	${AR} ${AROPTS} libtiff.a $?
 	${RANLIB} libtiff.a
 
-$(SHLIB): $(SHOBJS)
-	gcc -o $(SHLIB) -Wl,-soname,$(SONAME) -shared $(SHOBJS)
+$(SHLIB): @DSO@dso
 
 #
 # NB: The configure script verifies that the configured
@@ -351,7 +350,7 @@
 		-src $$i -O $$f;					\
 	done
 
-installDSO: @DSO@dso
+installDSO: $(SHLIB)
 	if [ @DSOSUF_VERSION@ != @DSOSUF@ ]; then			\
 	    ${INSTALL} -idb tiff.sw.tools -m 555 -F @DIR_LIB@		\
 		-O libtiff.@DSOSUF_VERSION@;				\
--- tiff-v3.5.6-beta/tools/Makefile.in.orig	Thu Jun 14 20:57:19 2001
+++ tiff-v3.5.6-beta/tools/Makefile.in	Thu Jun 14 21:03:20 2001
@@ -47,7 +47,7 @@
 IPATH	= -I. -I${SRCDIR} -I${LIBDIR}
 CFLAGS	= @ENVOPTS@ ${COPTS} ${OPTIMIZER} ${IPATH}
 #
-TIFFLIB	= ${DEPTH}/libtiff/libtiff.so.3.5
+TIFFLIB	= ${DEPTH}/libtiff/libtiff.@DSOSUF_VERSION@
 LIBJPEG	= @LIBJPEG@
 LIBGZ	= @LIBGZ@
 LIBTIFF	= -L${DEPTH}/libtiff  -ltiff
