--- m4/acinclude.m4~	2005-12-06 11:40:01.000000000 +0000
+++ m4/acinclude.m4	2008-09-05 14:24:05.000000000 +0100
@@ -391,7 +391,7 @@
   [ax_cv_check_glut_libglut="no"
   ax_save_LIBS="${LIBS}"
   LIBS=""
-  ax_check_libs="-lglut32 -lglut"
+  ax_check_libs="-lglut"
   for ax_lib in ${ax_check_libs}; do
     if test X$ax_compiler_ms = Xyes; then
       ax_try_lib=`echo $ax_lib | sed -e 's/^-l//' -e 's/$/.lib/'`
