--- tiff-v3.4/libtiff/Makefile.in.ewt	Tue Oct 14 21:16:40 1997
+++ tiff-v3.4/libtiff/Makefile.in	Tue Oct 14 21:51:36 1997
@@ -64,6 +64,8 @@
 COPTS	= @GCOPTS@
 OPTIMIZER=-O
 CFLAGS	= @ENVOPTS@ @LIBCOPTS@ ${COPTS} ${OPTIMIZER} ${IPATH} ${CONF_LIBRARY}
+SONAME = libtiff.so.3
+SHLIB = libtiff.so.3.4
 #
 SRCS	= \
 	tif_aux.c \
@@ -133,7 +135,8 @@
 	tif_write.o \
 	tif_zip.o \
 	${NULL}
-TARGETS	= libtiff.a
+TARGETS	= libtiff.a $(SHLIB)
+SHOBJS= $(OBJS:%.o=%.so) 
 
 all:	${TARGETS}
 	if [ @DSO@dso != nodso ]; then \
@@ -146,6 +149,9 @@
 	${AR} ${AROPTS} libtiff.a $?
 	${RANLIB} libtiff.a
 
+$(SHLIB): $(SHOBJS)
+	gcc -o $(SHLIB) -Wl,-soname,$(SONAME) -shared $(SHOBJS)
+
 #
 # NB: The configure script verifies that the configured
 # tools are capable of producing a DSO before enabling
@@ -328,3 +334,6 @@
 	    mkg3states tif_fax3sm.c \
 	    mkversion version.h \
 	    libtiff.a libtiff.@DSOSUF@ libtiff.@DSOSUF_VERSION@ *dso
+
+%.so : %.c
+	$(CC) -c -fPIC $(CFLAGS) $(CPPFLAGS) $< -o $@
--- tiff-v3.4/tools/Makefile.in.ewt	Tue Oct 14 21:55:37 1997
+++ tiff-v3.4/tools/Makefile.in	Tue Oct 14 21:56:06 1997
@@ -47,7 +47,7 @@
 IPATH	= -I. -I${SRCDIR} -I${LIBDIR}
 CFLAGS	= @ENVOPTS@ ${COPTS} ${OPTIMIZER} ${IPATH}
 #
-LIBTIFF	= ${DEPTH}/libtiff/libtiff.@DSOSUF@
+LIBTIFF	= ${DEPTH}/libtiff/libtiff.so.3.4
 LIBJPEG	= @LIBJPEG@
 LIBGZ	= @LIBGZ@
 LIBS	= ${LIBTIFF} ${LIBJPEG} ${LIBGZ} @LIBPORT@ @MACHDEPLIBS@
